ledger/complete/wal/wal.go

```diff
@@ -230,6 +248,29 @@ func (w *LedgerWAL) replay(
 	return nil
 }
 
+func getPossibleCheckpoints(allCheckpoints []int, from, to int) []int {
+	//list of checkpoints is sorted
+	indexFrom := sort.SearchInts(allCheckpoints, from)
+	indexTo := sort.SearchInts(allCheckpoints, to)
+
+	// all checkpoints are earlier, return last one
+	if indexTo == len(allCheckpoints) {
+		return allCheckpoints[indexFrom:indexTo]
+	}
+
+	// exact match
+	if allCheckpoints[indexTo] == to {
+		return allCheckpoints[indexFrom : indexTo+1]
```

Comment:
Does this mean we include an extra element when there's an exact match?

Reply:
Upper-bound is open, so it wouldn't be included without +1